src/constraint_trees.jl

```diff
@@ -63,9 +63,9 @@ ConstraintTrees.collect_variables!(x::ParameterQuadraticValue, out) =
     end
 
 ConstraintTrees.renumber_variables(x::ParameterLinearValue, mapping) =
-    ParameterLinearValue(idxs = [mapping[idx] for idx in x.idxs], weights = x.weights)
+    ParameterLinearValue(idxs=[mapping[idx] for idx in x.idxs], weights=x.weights)
 
-ConstraintTrees.renumber_variables(x::ParameterLinearValue, mapping) = ParameterLinearValue(
-    idxs = [(mapping[idx], mapping[idy]) for (idx, idy) in x.idxs],
-    weights = x.weights,
+ConstraintTrees.renumber_variables(x::ParameterQuadraticValue, mapping) = ParameterQuadraticValue(
+    idxs=[(mapping[idx], mapping[idy]) for (idx, idy) in x.idxs],
+    weights=x.weights,
```

Comment:
why remove the overload for ParameterLinearValue?

Reply:
can you pls share teh warning? (it should not.)

Reply:
```
Precompiling DifferentiableMetabolism...
Info Given DifferentiableMetabolism was explicitly requested, output will be shown live 
WARNING: Method definition renumber_variables(DifferentiableMetabolism.ParameterLinearValue, Any) in module DifferentiableMetabolism at /home/hettie/.julia/packages/DifferentiableMetabolism/wIzFd/src/constraint_trees.jl:65 overwritten at /home/hettie/.julia/packages/DifferentiableMetabolism/wIzFd/src/constraint_trees.jl:68.
ERROR: Method overwriting is not permitted during Module precompilation. Use `__precompile__(false)` to opt-out of precompilation.
  ? DifferentiableMetabolism
[ Info: Precompiling DifferentiableMetabolism [9677e053-0fc1-49e6-98a8-2ad3a1525e54] (cache misses: wrong dep version loaded (2), wrong source (2), dep missing source (4))
WARNING: Method definition renumber_variables(DifferentiableMetabolism.ParameterLinearValue, Any) in module DifferentiableMetabolism at /home/hettie/.julia/packages/DifferentiableMetabolism/wIzFd/src/constraint_trees.jl:65 overwritten at /home/hettie/.julia/packages/DifferentiableMetabolism/wIzFd/src/constraint_trees.jl:68.
ERROR: Method overwriting is not permitted during Module precompilation. Use `__precompile__(false)` to opt-out of precompilation.
┌ Info: Skipping precompilation due to precompilable error. Importing DifferentiableMetabolism [9677e053-0fc1-49e6-98a8-2ad3a1525e54].
└   exception = Error when precompiling module, potentially caused by a __precompile__(false) declaration in the module.
```

Reply:
should i keep two method definitions `renumber_variables(DifferentiableMetabolism.ParameterLinearValue, Any)` ..?